server/api.py

```diff
@@ -1446,6 +1447,72 @@ def create_tournament(
     return 200, tournament
 
 
+@api.put(
+    "/tournament/{tournament_id}/register-team",
+    response={200: TournamentSchema, 400: Response, 401: Response},
+)
+def add_team_registration(
+    request: AuthenticatedHttpRequest,
+    tournament_id: int,
+    team_details: AddOrRemoveTeamRegistrationSchema,
+) -> tuple[int, Tournament] | tuple[int, message_response]:
+    try:
+        team = Team.objects.get(id=team_details.team_id)
+    except Team.DoesNotExist:
+        return 400, {"message": "Team does not exist"}
+
+    if request.user not in team.admins:
+        return 401, {"message": "Only team admins can register a team to a tournament !"}
+
+    try:
+        tournament = Tournament.objects.get(id=tournament_id)
+    except Tournament.DoesNotExist:
+        return 400, {"message": "Tournament does not exist"}
+
+    if tournament.status != Tournament.Status.REGISTERING:
```

Comment:
Can we also check tournament type and team category - opens, womens etc after the PR merge before registering a team if they match the category and are eligible